var/spack/repos/builtin/packages/maqao/package.py

```diff
@@ -0,0 +1,132 @@
+# Copyright 2013-2022 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from llnl.util.filesystem import find_system_libraries
+
+from spack.package import *
+from spack.util.executable import *
+
+
+class Maqao(CMakePackage):
+    """MAQAO (Modular Assembly Quality Analyzer and Optimizer) is a performance
+    analysis and optimization framework operating at binary level with a focus
+    on core performance. Its main goal is to guide application developers
+    along the optimization process through synthetic reports and hints."""
+
+    homepage = "https://maqao.exascale-computing.eu/"
+    git = "https://gitlab.exascale-computing.eu/MAQAO/MAQAO.git"
+    url = "https://gitlab.exascale-computing.eu/MAQAO/MAQAO/repository/archive.tar.bz2?ref=master"
+
+    maintainers = ["cvalensi", "yaspr"]
+    executables = ["maqao"]
+
+    version("master", branch="master")
+
+    depends_on("cmake@2.8.12:", type="build")
+    depends_on('gcc@4.8.1: languages="c,c++"', when="%gcc")
+
+    # From 'profiles' subdirectory
+    variant(
+        "profile",
+        multi=False,
+        default="default",
+        values=("default", "release", "release.intel64-xeonphi"),
+        description="What profile to build",
+    )
+
+    # From 'src/arch.h' source file
+    variant(
+        "arch",
+        multi=True,
+        default="x86_64",
+        values=("ia32", "x86_64", "k1om", "arm", "arm64", "power", "thumb"),
+        description="What architectures to build",
+    )
```

Comment:
Can't this be inferred by the value of `target`?

Reply:
Yeah, that can be done with a dict to map each `target` to the corresponding arch code in the sources

Reply:
Update from the devs:
>The architectures currently supported by the released version of MAQAO 
are indeed limited to Intel64 and Xeon Phi. We have however defined a 
code for most architectures in case of future support (even if limited).